### ISSUE TITLE
ReadTheDocs build ends with error

### ISSUE THREAD

**Issue by marcus-oscarsson** [mxcube/mxcube3#1519]
The build process on ReadTheDocs ends with the following error message:
```
...
  - Installing sphinx-last-updated-by-git (0.3.8)
  - Installing sphinxcontrib-httpdomain (1.8.1)
  - Installing tzlocal (4.3.1)

Installing the current project: mxcubeweb (4.249.0)

[rtd-command-info] start-time: 2024-12-19T13:28:40.394331Z, end-time: 2024-12-19T13:58:00.962292Z, duration: 1760, exit-code: 137
mamba run --name mxcubeweb python -m sphinx -T -E -b html -d _build/doctrees -c docs docs/source ${READTHEDOCS_OUTPUT}/html
Command killed due to timeout or excessive memory consumption
```

It seems like the conda build/install consumes too much memory

**Comment by marcus-oscarsson** [comment]
The docs on http://mxcube.github.io/mxcubeweb/ is still built

**Comment by fabcor-maxiv** [comment]
Still stuck today. I changed the [branch protection rules for `develop`](https://github.com/mxcube/mxcubeweb/settings/branches) so that Read The Docs build is not "required" for now, we should revert once this is solved, but we have the GitHub Pages build anyway which is more or less equivalent.

**Comment by marcus-oscarsson** [comment]
Ok

**Comment by marcus-oscarsson** [comment]
The `read-the-docs` builds are still failing. I think I will disable the action and if possible redirect the `read-the-docs` page to the `GitHub` pages. What do you think ?

**Comment by fabcor-maxiv** [comment]
Another one of those things that is on my TO-DO list, but that keeps getting postponed... :/

I think I have an idea. I want to get rid of the "[REST API](https://mxcubeweb.readthedocs.io/en/stable/rest-api/index.html)" part of the documentation. Maybe this will help, maybe we need to get rid of more things.

**Comment by fabcor-maxiv** [comment]
It seems to me like it is starting with this PR that the Readthedocs builds started failing: "[Adding documentation and dependency for missing video stream #1504](https://github.com/mxcube/mxcubeweb/pull/1504)" where we added `ffmpeg` as a dependency in `conda-environment.yml`.

**Comment by fabcor-maxiv** [comment]
> It seems to me like it is starting with this PR that the Readthedocs builds started failing: "[Adding documentation and dependency for missing video stream #1504](https://github.com/mxcube/mxcubeweb/pull/1504)" where we added `ffmpeg` as a dependency in `conda-environment.yml`.

Yep... looks like removing `ffmpeg` from `conda-environment.yml` allows Readthedocs to build successfully and quickly...

I will think of a proper fix.

**Comment by marcus-oscarsson** [comment]
Thanks, great !

**Comment by walesch-yan** [comment]
> > It seems to me like it is starting with this PR that the Readthedocs builds started failing: "[Adding documentation and dependency for missing video stream #1504](https://github.com/mxcube/mxcubeweb/pull/1504)" where we added `ffmpeg` as a dependency in `conda-environment.yml`.
> 
> Yep... looks like removing `ffmpeg` from `conda-environment.yml` allows Readthedocs to build successfully and quickly...
> 
> I will think of a proper fix.

Would it make sense to use a lighter version of the `conda-environment.yaml` specifically for Readthedocs? I assume that it doesn't need most dependencies. Hence, the extra work for updating both would remain small?

I don't know if it is possible to create a lighter version of `conda-envrionment.yml` at run time. (Maybe with `conda-lock` and filtering?)

**Comment by fabcor-maxiv** [comment]
> Would it make sense to use a lighter version of the `conda-environment.yaml` specifically for Readthedocs? I assume that it doesn't need most dependencies. Hence, the extra work for updating both would remain small?

My plan is to check if we need any Conda YAML file (we could list the dependencies on the command line in the Readthedocs YAML file directly) or even if we need a Conda environment at all.

But yes, working with the `conda-environment.yml` directly would be better, because we would avoid duplicating lists of dependencies all over the place.

**Comment by fabcor-maxiv** [comment]
> I don't know if it is possible to create a lighter version of `conda-envrionment.yml` at run time. (Maybe with `conda-lock` and filtering?)

@walesch-yan, I am not familiar with `conda-lock`, I would not really know where to start. I see there is a concept of "[categories](https://conda.github.io/conda-lock/src_environment_yml/#categories)", maybe this is something we could use. Although, I am worried about introducing yet another file (among `conda-environment.yml`, `poetry.lock`, `pyproject.toml`, and `requirements.txt`). Do you have any suggestions?

**Comment by walesch-yan** [comment]
> > I don't know if it is possible to create a lighter version of `conda-envrionment.yml` at run time. (Maybe with `conda-lock` and filtering?)
> 
> [@walesch-yan](https://github.com/walesch-yan), I am not familiar with `conda-lock`, I would not really know where to start. I see there is a concept of "[categories](https://conda.github.io/conda-lock/src_environment_yml/#categories)", maybe this is something we could use. Although, I am worried about introducing yet another file (among `conda-environment.yml`, `poetry.lock`, `pyproject.toml`, and `requirements.txt`). Do you have any suggestions?

@fabcor-maxiv, honestly I'm also not too familiar with it, it is mainly used to pin specific versions for `conda` environments. But you can generate the files from existing `conda-environments.yml` files. During the installation part I think you can then set a flag with the dependencies you would like to install/skip (This is the part I'm not too sure about if it will work). 

So my idea would be to specify the packages needed from the conda environment directly in `.readthedocs.yaml`, `conda-lock` will then generate a file with the current versions from our main `conda-environment.yml` and we can then install only the necessary dependencies. That way we would add a new file, but it would be one, we would not have to keep up to date, as it's generation would happen during the PR request and will therefore always be in synch with `conda-environment.yml`. 

However, as I said it is only an idea and I'm not sure if it works, but if it is fine with you, I can give it a try and investigate a little about this?:)

**Comment by walesch-yan** [comment]
My bad, I checked the documentation more thoroughly and unfortunately, this functionality (which I thought would exist) does not seem to exist for `conda-lock`.

**Comment by marcus-oscarsson** [comment]
Hm, not an easy problem to solve elegantly I guess. How about a mix between the current solution proposed and using `conda-environment,yml`. As @walesch-yan pointed out there is a risk that we forget or need to update the version of python in several places when changing python version.  So, perhaps, what we could do is to grep/sed the python version from the `conda-environment.yml` and add it to the `mamba env create --name mxcubeweb ....` line. 

Like this we have don't have to maintain several files and we don't need to edit in several places.

**Comment by fabcor-maxiv** [comment]
OK, I got it to build the docs with <https://github.com/mxcube/mxcubeweb/pull/1533>. That is the best fix I can pull off for now. :)
